src/js/controller/LayersListController.js

```diff
@@ -123,14 +124,50 @@
     }
   };
 
-  ns.LayersListController.prototype.renameCurrentLayer_ = function () {
+  ns.LayersListController.prototype.focusLayerElement_ = function(element) {
+    var index = pskl.utils.Dom.getData(element, 'layerIndex');
+    this.piskelController.setCurrentLayerIndex(parseInt(index, 10));
+
+    var selectedClass = 'current-layer-item';
```

Comment:
Even if we change the RESET event, the point of this comment was to say: there should be only one way to set the "current-layer-item" class on the selected element. 

Here the new code does it by DOM manipulation, while we used to set it by re-rendering the template before. I think we should remove the part of the template that can set this classname, isolate the code your new code to a dedicated method and also call it from renderLayersList_